diff --git a/variants/heltec_v4/HeltecV4Board.cpp b/variants/heltec_v4/HeltecV4Board.cpp
index 3f13f41..7f0ae67 100644
--- a/variants/heltec_v4/HeltecV4Board.cpp
+++ b/variants/heltec_v4/HeltecV4Board.cpp
@@ -41,18 +41,18 @@ void HeltecV4Board::begin() {
   }
 
   uint16_t HeltecV4Board::getBattMilliVolts()  {
-    analogReadResolution(10);
+    analogReadResolution(12);
     digitalWrite(PIN_ADC_CTRL, HIGH);
     delay(10);
     uint32_t raw = 0;
     for (int i = 0; i < 8; i++) {
-      raw += analogRead(PIN_VBAT_READ);
+      raw += analogReadMilliVolts(PIN_VBAT_READ);
     }
     raw = raw / 8;
 
     digitalWrite(PIN_ADC_CTRL, LOW);
 
-    return (adc_mult * (3.3 / 1024.0) * raw) * 1000;
+    return adc_mult * raw;
   }
 
   const char* HeltecV4Board::getManufacturerName() const {
